statistics/statistics.go

```diff
@@ -21,329 +21,143 @@ package statistics
-		case int32:
-			buffer = append(buffer, float64(val.(int32)))
-		case int64:
-			buffer = append(buffer, float64(val.(int64)))
-		case float64:
-			buffer = append(buffer, val.(float64))
-		case float32:
-			buffer = append(buffer, float64(val.(float32)))
-		case uint64:
-			buffer = append(buffer, float64(val.(uint64)))
-		case uint32:
-			buffer = append(buffer, float64(val.(uint32)))
-		}
-	}
+	policy.AddNewIntRule([]string{""}, "slidingWindowLength", false, plugin.SetDefaultInt(100), plugin.SetMinInt(1))
+	policy.AddNewIntRule([]string{""}, "slidingFactor", false, plugin.SetDefaultInt(1), plugin.SetMinInt(1))
+	policy.AddNewStringRule([]string{""}, "statistics", false, plugin.SetDefaultString(strings.Join(statList, ",")))
+	return *policy, nil
 
-	statList := [...]string{"count", "mean", "median", "standard_deviation", "variance", "95%_ile", "99%_ile", "2%_ile", "9%_ile", "25%_ile", "75%_ile", "91%_ile", "98%_ile", "minimum", "maximum", "range", "mode", "kurtosis", "skewness", "sum", "trimean", "quartile_range"}
-	min, minErr := stats.Min(buffer)
-	max, maxErr := stats.Max(buffer)
+}
 
-	for _, stat := range statList {
-		switch stat {
-		case "count":
-			val = float64(len(buffer))
-		case "mean":
-			val, err = stats.Mean(buffer)
-		case "median":
-			val, err = stats.Median(buffer)
-		case "standard_deviation":
-			val, err = stats.StandardDeviation(buffer)
-		case "variance":
-			val, err = stats.Variance(buffer)
-		case "95%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 95)
-		case "99%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 99)
-		case "2%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 2)
-		case "9%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 9)
-		case "25%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 25)
-		case "75%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 75)
-		case "91%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 91)
-		case "98%_ile":
-			val, err = stats.PercentileNearestRank(buffer, 98)
-		case "minimum":
-			val, err = min, minErr
-		case "maximum":
-			val, err = max, maxErr
-		case "range":
-			val = max - min
-		case "mode":
-			modeVal, err = stats.Mode(buffer)
-		case "kurtosis":
-			val, err = p.Kurtosis(buffer)
-		case "skewness":
-			val, err = p.Skewness(buffer)
-		case "sum":
-			val, err = stats.Sum(buffer)
-		case "trimean":
-			val, err = stats.Trimean(buffer)
-		case "quartile_range":
-			val, err = stats.InterQuartileRange(buffer)
-		default:
-			st := fmt.Sprintf("Unknown statistic received %T:", stat)
-			log.Errorf(st)
-			err = errors.New(st)
+// Process processes the data, inputs the data into sorted buffer and calls the GetStats method
+func (p *Plugin) Process(metrics []plugin.Metric, cfg plugin.Config) ([]plugin.Metric, error) {
+	var result []plugin.Metric
+	for _, metric := range metrics {
+		slidingWindowLength, slidingFactor, stats, err := GetConfig(metric.Config)
```

Comment:
are we using collector -> processor -> publisher in this order? What do you mean case by case. Be specific please.